modules/apps/layout/layout-page-template-service/src/main/java/com/liferay/layout/page/template/internal/upgrade/v1_1_0/UpgradeLayoutPrototype.java

```diff
@@ -97,14 +100,28 @@ protected void upgradeLayoutPrototype() throws Exception {
 					int i = 1;
 
 					while (true) {
-						String newName = name + StringPool.DASH + i;
+						String newName;
+
+						if (name.length() == 75) {
+							newName = name.substring(0, 73);
```

Comment:
If we want to cover all cases we shouldn't use 73 as limit for the name, picture that we get -10 or even -100 as a suffix. I know it is rare and it's not going to happen but the code has to cover 100% cases.

We should get the substring based on the suffix length.

Can you change it?